src/utils/editor/expression.js

```diff
@@ -46,7 +46,25 @@ export function updatePreview(
 
   const isUpdating = preview && preview.updating;
 
-  const inScope = linesInScope && linesInScope.includes(location.line);
+  let inScope;
+  const isLineInScope = linesInScope && linesInScope.includes(location.line);
+  if (isLineInScope) {
```

Comment:
lets move this into its own function which returns a value that can be assigned to `inScope`

here is how it would looks like:
```javascript
function isInScope(linesInScope, location) {
  const isLineInScope = linesInScope && linesInScope.includes(location.line);
  if (isLineInScope) {
    return true;
  }

  const startLineLocation = outOfScopeLocations.find(
    loc => loc.start.line === location.line
  );
  if (startLineLocation) {
    return startLineLocation.start.column > location.column;
  }
  
  const endLineLocation = outOfScopeLocations.find(
    loc => loc.end.line === location.line
  );
  if (endLineLocation) {
    return endLineLocation.end.column < location.column;
  }    

  // if none of these conditions are true, return false
  return false;
}

// use
const inScope = isInScope(linesInScope, location)
```

Reply:
love it